### PR TITLE
i ded pls nerf: punches cannot stun if the person is already on the ground

### PR DESCRIPTION
**Pull request by Iamgoofball**
Yes, I died, yes, I'm salty.

I feel like you shouldn't be able to knock someone down with punches if they're already on the ground. This should make unarmed combat less about "who got the punch stun first".

### CODE REVIEW

**Comment by optimumtact** [comment]
can you just not make hand2hand combat stun at all

**Comment by Iamgoofball** [comment]
now I'd do that but that'd be even worse when the greytiders see this PR, I snuck it in during a round restart(by accident) so they haven't noticed it yet

**Comment by MMMiracles** [comment]
Chain-weakening during unarmed combat was always really dumb so I don't' see any problems with this.
:+1:

**Comment by PKPenguin321** [comment]
I'm on the fence with this
on one hand, UNARMED MAN KILLS ENTIRE NUKE OPS SQUAD WITH BARE HANDS HOLY SHIT THIS GAME IS SO COOL
on the other hand, assistant vs assistant fistfights will be cooler sorta

if it turns out to be a bad change we can always revert it i guess
:+1: for now

**Comment by Anonus** [comment]
Nerfs golem punching significantly :+1:

**Comment by Razharas** [comment]
>Kick to face when downed wont keep you downed for longer

0/10

**Comment by Bawhoppen** [comment]
@PKPenguin321 thats what soap is for :leopard:

**Comment by Supermichael777** [comment]
take the rng off of punches altogether the rng is just to strong for somthing that you literally cannot take away from people.

**Comment by drynwyn** [comment]
No. People want a MORE lethal station, not less. The fact that even if you have the HOS coat and the entire armory in your bag you can still be taken down by a naked pink-haired man if you let your guard down is a critical part of SS13.

This would be a huge buff to all the jobs that can easily get stun weapons.

**Comment by MMMiracles** [comment]
You can still be taken down by a naked pink-haired man, its just that the naked pink-haired man won't be able to mercilessly chain-weaken you with literally nothing to counter it.

Are you really trying to say RNG weakening is a fun mechanic, let alone with it can literally stun-lock you with no hope of escape?

**Comment by ike709** [comment]
:+1: I never get the lucky stun punch anyways

File: code/datums/martial.dm [hunk 1]
```diff
@@ -51,7 +51,7 @@
 
 	add_logs(A, D, "punched")
 
-	if((D.stat != DEAD) && damage >= 9)
+	if(!D.stat && !D.paralysis && !D.stunned && !D.weakened && damage >= 9)
```
Comment (Aranclanos) [inline code/datums/martial.dm]: the code is horrible

**Comment by Cheridan** [comment]
I don't think this is good. It's the removal of yet another equalizer. People should be able to fight back and be a threat even if they don't have armor and stun weapons. 
If getting chainstunned is so bad, then why the hell are you starting with punches and not tasers or batons?

File: code/datums/martial.dm (hunk 1 above)
Comment (Iamgoofball) [inline code/datums/martial.dm]: Would a is_stunned() macro help here? Or do we already have one?

**Comment by Iamgoofball** [comment]
Because I don't want to get into that issue, this is just a fist fight thing. Besides, the lucky stun punch is still there, it just means you aren't as likely to get 3 stuns in a row on someone who's already on the ground, keeping them down for about as lonv as a stun baton stun.

**Comment by Fox-McCloud** [comment]
Goon did this a while back, but for pushing.

**Comment by drynwyn** [comment]
@MMMiracles RNG weakening is a good mechanic, because it can't "literally stun lock you with no hope of escape". A chain of lucky weakens can keep you down long enough to kill you, yeah, but a stunprod and fire extinguisher can do the same thing just as easily. Equalizers are good.

**Comment by Iamgoofball** [comment]
@Fox-McCloud I'll probably do that in another pull request then.

**Comment by KorPhaeron** [comment]
>armed with a taser, stun baton, flashbang, pepperspray, all reliable one hit stuns, two of which are AoE, one is ranged/spammable, one is melee and can be re-applied forever
>have the AI and rest of sec a headset call away
>somehow die anyway to someone unarmed

No amount of punching nerfs will save you

@MMMiracles 

Like Cheridan said, it's an equalizer. Naked clown chainstuns nuke op to death with his fists is more interesting than "oh that guy had better stuff I guess he wins no matter what now"

**Comment by Man-Shroom-** [comment]
yes :+1: to not being able to disarm-push someone onto the floor while they are on the floor, as stated by Fox up there

no :-1: to not being able to chain kick/punch. being stomped in the face should keep you on the ground and it is definitely not 'an unbreakable impossible-to-escape situation because obviously someone needs to be extremely lucky to chainstun a person to death. think about how many rounds of this game have you played, and how many times has this happened to you? it is very rare.

**Comment by MMMiracles** [comment]
This doesn't remove that 'holy shit' moment when you land that magical weaken on your opponent, it just makes it so you can't RNG chain-stun them with fists of fury (as funny as that may be).

I get there needs to be some sort of equalizer but nothings more salt-inducing then accepting death as mcgrayshirt proceeds to pull off 4-5 consecutive weaken stuns and cuck you.

**Comment by optimumtact** [comment]
it's extremely annoying RNG
I wouldn't be oppossed to kicks causing stam damage or something

but right now it's basically a toss up as to being absolutely pointless and gettign you killed or literally letting you ruin someone no matter how geared they are.

**Comment by Man-Shroom-** [comment]
it is incredibly rare for that to actually happen and someone needs to be incredibly lucky for that to happen

**Comment by optimumtact** [comment]
that's bullshit and you know it.

I've had it happen to me many times

**Comment by Man-Shroom-** [comment]
out of how many rounds that you have played? thousands? and it has happened 3-4 times?

**Comment by PKPenguin321** [comment]
i did some math

with some massive sample sizes (exactly 500 punches), i averaged 4.62 damage per punch, which goes up to about 5 damage
it would take 20 of these punches to drop an unarmored, default human into critical condition

with absolutely 0 lag (direct server connection), you can get 7 punches off on the immobile target if you click as fast as you can.
if you do exactly 5 damage per punch (which is actually probably kinda generous), you can deal 35 damage, plus the 9 damage from the weaken punch itself, equaling 41 damage.

realistically, this number is really give or take. you could do 6-8 damage with each punch, or you could whiff every punch after the weaken. punching is too random to get a real hard number on this. my average means practically nothing when punching itself is completely RNG, but i'm using it for the sake of being able to reasonably calculate the probability of killing with a stun punch. also note that actually getting the full 7 punches in after a weaken is unlikely in a real combat scenario. in my test, it was played completely lagless, as optimally as possible, with a completely unmoving target. **all of these factors mean that the actual damage off of a weaken punch ingame is much lower than my actual result, so my math here should be taken with a grain of salt.**

there is a 1 in 10 chance of getting a weaken punch. to put a default human into critical with my numbers, you would need 2.44 stun punches, or just two weakens and another two 5~ damage punches. the probability of getting 2.44 stun punches in a row is, if i did this right, (1/10)^2.44, which is *really* tiny. that means with absolutely optimal, perfect play, you still only have a minuscule chance of killing with exclusively punches.

tl;dr: Punches not worthy of nerfing

**Comment by Bawhoppen** [comment]
@PKPenguin321 :jp:

**Comment by MMMiracles** [comment]
did you account in golems because their punchmod makes them hilariously lethal in fist fights due to their extra damage.

**Comment by PKPenguin321** [comment]
who gives a fuck about golems
if you want to nerf golems, change their punchmod

**Comment by KorPhaeron** [comment]
>golems

Just walk away slowly?

**Comment by MMMiracles** [comment]
walk away from the golem swarm that is xenobio.

no escaping the golemtide.

**Comment by Bawhoppen** [comment]
if xenobio is well developed, theres bigger things to worry about than golems

mainly infinite deathcarps, infinite change carps, infinite heal carps, infinite teleport carps, 
syndieborgs, hivebot swarm, sentient bots with infinite stunning, infinite revolvers, xeno swarm, etc.

**Comment by MMMiracles** [comment]
last time i checked everything listed there is extremely RNG-based in terms of getting and some aren't even possible without admin intervention (mostly xenos).

**Comment by Bawhoppen** [comment]
everything I listed you can get in xenobio in 30 minutes

There is some RNG involved but the main part of RNG is the slime splitting itself

My point is the golems are not a problem.

**Comment by MMMiracles** [comment]
its a hell of a lot easier to mass produce golems than it is anything else you listed, though.

**Comment by Bawhoppen** [comment]
when you get your first chaos carp or magicarp with SoC everything is about 100x easier to produce because its just 'left click until you get what you want'

**Comment by optimumtact** [comment]
your evaluation is disingenuous because it misses the root of the problem, which is that you usually have a deadly weapon in your hand and you drop it when you get stunned. leaving the person free to murder you with your own weapon.

I don't think the problem is chainstunning, I think it's the fact it stuns you randomly at all. if used stamina or otherwise had mechanics that made sense it wouldn't be an issue.

Disarm has the same issue on the pushdown stun.

**Comment by KorPhaeron** [comment]
> which is that you usually have a deadly weapon in your hand

So shoot/hit them with it before they punch you

**Comment by optimumtact** [comment]
THANKS FOR THE ADVICE KOR

File: code/datums/martial.dm (hunk 1 above)
Comment (phil235) [inline code/datums/martial.dm]: Unfortunately we can't use !D.incapacitated() since it also check for restrained() (i.e. handcuffed)

**Comment by Man-Shroom-** [comment]
hey guys let's nerf punches for all humans because golems are heavy hitters despite being slower the internet in the pitcairn islands

**Comment by Iamgoofball** [comment]
This wasn't about golems.

**Comment by Robustin** [comment]
Mixed feelings on this, on one hand hand-to-hand used to be shit, on the other hand I know the feeling of being a traitor with a murder arsenal in your bag and one fucknugget sees you slip c4 into your pocket or something and chain weakens you to death. On the other other hand some pissy assistant punching a traitor to death is funny.

Overall I say leave it, this game is better when random idiots have a chance to beat the shit out of people who vastly outgun them.

**Comment by tkdrg** [comment]
I've never had any issue with this. RNG is cool.

**Comment by PKPenguin321** [comment]
close this PR when

**Comment by Cheridan** [comment]
git gud
